app/api-client.js

```diff
@@ -1,170 +1,291 @@
-const config = require('../app/config');
+const config = require('./config');
 const request = require('request-promise');
 const url = require('url');
 
 
-module.exports = (function () {
-  var token = 'invalid token';
+const log = require('bole')('api-client');
+
+
+class JWTAuth {
+  constructor(token) {
+    this.token = token || 'invalid token';
+  }
+
+  set_token(token) {
+    this.token = token;
+  }
+
+  unset_token() {
+    this.token = 'invalid token';
+  }
+
+  get header() {
+    return `JWT ${this.token}`;
+  }
+}
+
+
+class APIClient {
+  constructor(conf) {
+    this.base_url = conf.base_url;
+    this.auth = null;
+  }
+
+  request(endpoint, method = 'GET', body = null) {
+    const headers = {};
```

Comment:
Watch out here as you're declaring this as a constant but adding to it later.

I think this may still be possible because of the way JS objects work but it still seems weird.

Reply:
`const` means you can't reassign to the same variable name, not that the value is immutable.